view/authentication/login/app.js

```diff
@@ -55,7 +72,7 @@ function sendData(jsonData) {
       document.getElementById('errorInfo').innerHTML = errorText;
     } else {
       returnUrl = getQueryParams(window.location.search.substring(1));
-      if (returnUrl.ru) {
+      if (isValidURL(returnUrl.ru)) {
```

Comment:
It might be useful to add some unit tests for getQueryParams because the correctness of that implementation will directly impact the correctness of the isValidUrl implementation

Reply:
I have a quick look at `getQueryParams`, it's probably not an elegant way to extract the query param, but it will work in the use case here, the correctness depends on the correctness of the previous redirect. If the queryParam is manually input and it should consider invalid, anyway. 
I would keep the change of the scope small for now (instead of refactoring it).